src/hooks.ts

```diff
@@ -69,12 +66,19 @@ export function useCallsDataSubscription(
       const data = result?.data && result.data !== '0x' ? result.data : undefined
       results.push({ valid: true, data, blockNumber: result?.blockNumber })
     }
+    return results
+  }, [callResults, calls, chainId])
 
-    if (!areCallResultsEqual(results, stableResults.current)) {
-      stableResults.current = results
+  // Force the results to be referentially stable if they have not changed.
+  // This is necessary because *all* callResults are passed as deps when initially memoizing the results.
+  const lastResults = useRef(results)
+  return useMemo(() => {
```

Comment:
question: any particular reason behind splitting into two separate `useMemo`s? AFAIK this will run each time the previous `useMemo` is run since it always recreates a `results` array.

Reply:
btw, I am not sure whether there are any performance challenges related to the use of this hook in particular, but... since we're already constructing a results array on line 67, we could check for equality with `lastResults` there instead. 

Would reduce number of loops to just one, but again, just bringing it up more like a comment than actual actionable thing.

Reply:
I thought this was more easily readable code - that's the only reason. It _will_ be less performant, but I didn't know that that was an issue.